### PR TITLE
Simplify gin config for ac, ppo, ddpg and sarsa

### PR DESCRIPTION
**Pull request by emailweixu**
so that we don't need to set specs for various networks in the gin config.
TODO for sac since it's quite complicated.

### CODE REVIEW

File: alf/algorithms/actor_critic_algorithm.py [hunk 1]
```diff
@@ -70,6 +74,10 @@ def __init__(self,
             debug_summaries (bool): True if debug summaries should be created.
             name (str): Name of this algorithm.
             """
+        actor_network = actor_network_ctor(
+            input_tensor_spec=observation_spec, action_spec=action_spec)
+        value_network = value_network_ctor(input_tensor_spec=observation_spec)
+
         super(ActorCriticAlgorithm, self).__init__(
             observation_spec=observation_spec,
```
Comment (hnyu) [inline alf/algorithms/actor_critic_algorithm.py:82]: I feel that it's unnecessary to pass observation_spec to the base RLAlgorithm class.

**Review by hnyu — CHANGES_REQUESTED** [review]
Some PPO gin files have not been updated

**Comment by emailweixu** [comment]
> Some PPO gin files have not been updated

Those ppo gin files are not converted from tf version yet.

File: alf/algorithms/actor_critic_algorithm.py (hunk 1 above)
Comment (emailweixu) [reply alf/algorithms/actor_critic_algorithm.py:82]: It doesn't hurt to leave it there. It was once useful.

**Review by Haichao-Zhang — APPROVED** [review]

**Review by hnyu — APPROVED** [review]

**Pull request merged**